src/analyses/basePriv.ml

```diff
@@ -117,63 +117,80 @@ struct
   let lock ask getg st m = st
   let unlock ask getg sideg st m = st
 
-  let escape ask getg sideg st escaped = st
-  let enter_multithreaded ask getg sideg st = st
-  let threadenter = old_threadenter
-  let threadspawn ask getg sideg st = st
-
-  let iter_sys_vars getg vq vf =
-    match vq with
-    | VarQuery.Global g -> vf g
-    | _ -> ()
-
-
-  let read_global ask getg (st: BaseComponents (D).t) x =
+  let read_global (ask: Queries.ask) getg (st: BaseComponents (D).t) x =
     getg x
 
-  let write_global ?(invariant=false) ask getg sideg (st: BaseComponents (D).t) x v =
-    if invariant then (
-      (* Do not impose invariant, will not hold without privatization *)
-      if M.tracing then M.tracel "set" ~var:x.vname "update_one_addr: BAD! effect = '%B', or else is private! " (not invariant);
-      st
-    )
-    else (
-      (* Here, an effect should be generated, but we add it to the local
-      * state, waiting for the sync function to publish it. *)
-      (* Copied from MainFunctor.update_variable *)
-      if ((get_bool "exp.volatiles_are_top") && (is_always_unknown x)) then
-        {st with cpa = CPA.add x (VD.top ()) st.cpa}
```

Comment:
Why can these checks be removed?

Reply:
That's a good question! None of the other privatizations actually do anything special for `exp.volatiles_are_top` either. It could be that the option has been broken for years now anyway.

If that's the case and we want to fix it (or remove it), then I'd say it's a separate matter to be dealt with. Either way, I don't think this is something that every privatization needs to handle itself, but that should be somewhere higher.

Reply:
Since we're touching this right now, it would be good to leave it in for now and maybe make an issue to check what the others do?

Reply:
I added it back. I thought about opening a new issue,  but then found #810 which can be considered to also include these matters.